dagshub/common/cli.py

```diff
@@ -105,8 +114,77 @@ def upload(ctx,
 
     owner, repo_name = repo
     repo = Repo(owner=owner, name=repo_name, branch=branch)
-    repo.upload(file=filename, path=target, commit_message=message, force=update)
+    try:
+        repo.upload(file=filename, path=target, commit_message=message, force=update)
+    except RuntimeError as re:
+        # if uploaded filename already exists but update is False, a strange error message will appear
+        # since gogs tries to check last commit in this case
+        if "invalid last_commit" in str(re):
+            raise RuntimeError('Verify that the uploaded file is new') from re
+        else:
+            raise re
+
+
+@cli.command()
+@click.argument("repo", callback=validate_repo)
+@click.option("-u", "--upload-data", help="Upload data from specified url to new repository")
+@click.option("-c", "--clone", is_flag=True,  help="Clone repository locally")
+@click.option("-v", "--verbose", default=0, count=True, help="Verbosity level")
+def repo_create(
+                repo,
+                upload_data,
+                clone,
+                verbose):
+    """
+    repo_create
+    example:  dagshub repo-create sdafni.yd/yuvtutorial14 -u "http://0.0.0.0:8080/index.html" --checkout
+    """
+    logger = logging.getLogger()
+    logger.setLevel(to_log_level(verbose))
+
+    owner, repo_name = repo
+    INITIAL_DATA_FILE_NAME = 'repo_created_initial_data'
+
+    #
+    # create the repo in dagshub
+    #
+    repo = create_repo(repo_name)
+    logger.info(f"Created repo: {DEFAULT_HOST}/{owner}/{repo_name}.git")
```

Comment:
We need to have an ability for users to override the host and not only do it with dagshub.com
You can get it from the cli's context `ctx` (need to specify the ctx argument and add a decorator. You can look at the other functions to see how they do it)

Reply:
done (pending push)